### PR TITLE
fix imports & merge azure workflows

### PR DESCRIPTION
**Pull request by Borda**
<details>
  <summary><b>Before submitting</b></summary>

- [x] Was this discussed/agreed via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the [contributor guideline](https://github.com/Lightning-AI/pytorch-lightning/blob/main/.github/CONTRIBUTING.md), Pull Request section?
- [ ] Did you make sure to update the docs?
- [ ] Did you write any new necessary tests?

</details>

## What does this PR do?

merge duplicate workflows and add testing again stable branch

## PR review

Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in GitHub issues there's a high chance it will not be merged.

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

**Comment by Borda** [comment]
Thinking about enabling starting of multiple agents on a single machine...
@jerome-habana do you think there can be some collision of multiple users using the came machine at the same time? :thinking:

**Comment by Borda** [comment]
seems like all jobs hang, could it be because they really need to be run separately?

**Comment by jerome-habana** [comment]
> Thinking about enabling starting of multiple agents on a single machine... @jerome-habana do you think there can be some collision of multiple users using the came machine at the same time? 🤔

@Borda its needs to be executed separately as there is a possibility of root rank being shared across processes.

**Review by justusschock — APPROVED** [review]

**Review by justusschock — APPROVED** [review]

File: src/lightning_habana/fabric/strategies/single.py [hunk 1]
```diff
@@ -14,11 +14,23 @@
...
 from typing import Any, Dict, Optional
 
-from lightning.fabric.accelerators import Accelerator
-from lightning.fabric.plugins import CheckpointIO
-from lightning.fabric.plugins.io.torch_io import TorchCheckpointIO
-from lightning.fabric.strategies.single_device import SingleDeviceStrategy
-from lightning.fabric.utilities.types import _DEVICE, Optimizable
+from lightning_utilities import module_available
+
+if module_available("lightning"):
```
[the remark this answers is not in the document]
Comment (Borda) [reply src/lightning_habana/fabric/strategies/single.py:19]: Correct, but that is another topic...
we may want to add an env variable with setting priority importing

[the remark this answers is not in the document]
Comment (Borda) [reply src/lightning_habana/fabric/strategies/single.py:19]: @justusschock could you pls open an issue here to raise this concern?

**Pull request merged**